classes/Link.php

```diff
@@ -803,55 +805,34 @@ public function getAdminLink($controller, $withToken = true, $sfRouteParams = ar
                     // 'AdminRequestSql' => 'admin_sql_request', @todo: uncomment when CQRS pages are hookable
                     'AdminMeta' => 'admin_meta',
                     // 'AdminWebservice' => 'admin_webservice', @todo: uncomment when grid and entity form are done.
-                    'AdminBackup' => 'admin_backup',
                 );
 
                 if (isset($routes[$controller])) {
                     $routeName = $routes[$controller];
                 }
         }
 
-        if (isset($sfRouter)) {
-            if (empty($routeName)) {
-                $routeName = $this->searchRouteFromRouter($sfRouter, $controller);
+        if (empty($routeName) && null !== $legacyUrlConverter) {
+            try {
+                $conversionParameters = array_merge(['controller' => $controller], $sfRouteParams, $params);
+                unset($conversionParameters['token']);
+
+                return $legacyUrlConverter->convertByParameters($conversionParameters);
+            } catch (CoreException $e) {
```

Comment:
Add a comment to say you don't care if an exception occurred here.

Reply:
done